core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -88,10 +87,29 @@
     boolean isEmpty();
 
     /**
-     * Returns a {@link CompletableFuture} that completes when this publisher is complete,
+     * Return {@code true} if this stream is complete, either successfully or exceptionally,
```

Comment:
Thanks, the naming was driving me a bit crazy.

Possibly here or elsewhere, it'd be good to define `complete`, e.g., publisher has indicated it has finished writing by calling `close()` or `abort()`, and subscriber has either consumed all writes or cancelled.

Reply:
Updated Javadoc. A little bit of duplication but better being user-friendly :-)